src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -340,13 +341,27 @@ private void doRetrieve(SCMHeadObserver observer, TaskListener listener, GHRepos
             }
             @Override public boolean exists(@Nonnull String path) throws IOException {
                 try {
-                    repo.getFileContent(path, ref);
-                    listener.getLogger().format("      %s exists in this %s%n", path, thing);
-                    return true;
-                } catch (FileNotFoundException x) {
-                    listener.getLogger().format("      %s does not exist in this %s%n", path, thing);
+                    int index = path.lastIndexOf('/') + 1;
+                    List<GHContent> directoryContent = repo.getDirectoryContent(path.substring(0, index), ref);
+                    for (GHContent content : directoryContent) {
+                        if (content.isFile()) {
+                            String filename = path.substring(index);
+                            if (content.getName().equals(filename)) {
+                                listener.getLogger().format("      ‘%s’ exists in this %s%n", path, thing);
+                                return true;
+                            }
+                            if (content.getName().equalsIgnoreCase(filename)) {
+                                listener.getLogger().format("      ‘%s’ not found (but found ‘%s’, search is case sensitive) in this %s, skipping%n", path, content.getName(), thing);
+                                return false;
+                            }
+                        }
+                    }
+                } catch (FileNotFoundException fnf) {
+                    listener.getLogger().format("      ‘%s’ does not exist in this %s%n", path, thing);
                     return false;
```

Comment:
Would suffice to delete both lines and let execution fall through to the end of the method.